lib/elixir/lib/kernel.ex

```diff
@@ -4956,6 +4955,10 @@ defmodule Kernel do
   defp expand_module_alias(other, env), do: Macro.expand(other, env)
 
   # defmodule Elixir.Alias
+  defp alias_defmodule({:__aliases__, _, [:"Elixir", _]}, module, _env),
+    do: {module, module, false}
```

Comment:
I'm abusing the fact that both `nil` and `false` are both invalid module names since I needed to make the distinction, but maybe I should just match on `{:__aliases__, _, [:"Elixir", _]` in the `case`?